tests/testthat/test-linear_pool.R

```diff
@@ -453,13 +583,211 @@ test_that("samples only collected and re-indexed for simplest case", {
+    component_model = letters[1:6],
+    distinct_ids = 1
+  )
+  expected_outputs$component_model[expected_outputs$location == "222" &
+                                     expected_outputs$component_model %in% letters[5:6]] <- models_to_resample
+  expected_outputs$component_model[expected_outputs$location == "888" &
+                                     expected_outputs$component_model %in% letters[4:6]] <- letters[1:3]
+  expected_outputs <- expected_outputs |>
+    dplyr::arrange(dplyr::across(c("location", "target", "target_date", "component_model"))) |>
+    dplyr::tibble()
+
+  set.seed(1234)
+  actual_outputs <- sample_outputs |>
+    dplyr::mutate(component_model = model_id, .before = 1) |>
+    linear_pool(
+      weights = NULL,
+      task_id_cols = c("target_date", "target", "horizon", "location"),
+      compound_taskid_set = c("target", "location", "target_date"),
+      n_output_samples = 6
+    ) |>
+    dplyr::group_by(dplyr::across(dplyr::all_of(
+      c("location", "target", "target_date", "output_type_id", "component_model")
+    ))) |>
+    dplyr::summarize(distinct_ids = dplyr::n()) |>
+    dplyr::ungroup() |>
+    dplyr::select(-"output_type_id")
+  expect_equal(actual_outputs, expected_outputs)
+})
+
+
+test_that("ensemble of samples correctly drawn for compound task ID sets", {
+  sample_outputs <- expand.grid(stringsAsFactors = FALSE,
+                                model_id = letters[1:4],
+                                location = c("222", "888"),
+                                horizon = 1, #week
+                                target = "inc death",
+                                target_date = as.Date("2021-12-25"),
+                                output_type = "sample",
+                                output_type_id = 1:3,
+                                value = NA_real_)
+
+  sample_outputs$value[sample_outputs$location == "222" &
+                         sample_outputs$output_type_id == 1] <-
+    c(40, 30, 45, 80)
+  sample_outputs$value[sample_outputs$location == "222" &
+                         sample_outputs$output_type_id == 2] <-
+    c(60, 40, 75, 20)
+  sample_outputs$value[sample_outputs$location == "222" &
+                         sample_outputs$output_type_id == 3] <-
+    c(10, 70, 15, 50)
+  sample_outputs$value[sample_outputs$location == "888" &
+                         sample_outputs$output_type_id == 1] <-
+    c(100, 325, 400, 300)
+  sample_outputs$value[sample_outputs$location == "888" &
+                         sample_outputs$output_type_id == 2] <-
+    c(250, 350, 500, 250)
+  sample_outputs$value[sample_outputs$location == "888" &
+                         sample_outputs$output_type_id == 3] <-
+    c(150, 300, 500, 350)
+  sample_outputs <- sample_outputs |>
+    dplyr::mutate(horizon = 0, value = 0.75 * value) |>
+    dplyr::bind_rows(sample_outputs)
+
+  # All compound units have unique ids which are shared across dependent task columns
```

Comment:
What do you mean by "All compound units have unique ids" -- do you mean `output_type_id`s?

Reply:
Yes — I've updated the in-line comments to be clearer